lib/ex_check/check.ex

```diff
@@ -5,7 +5,7 @@ defmodule ExCheck.Check do
   alias __MODULE__.{Compiler, Pipeline}
 
   def run(opts) do
-    {tools, config_opts} = Config.load()
+    {tools, config_opts} = Config.load(opts)
```

Comment:
Since after suggestion below you'll probably end up with `:config` option, I'd rename it to `:file` when passing to loader in order to keep the name relevant & specific in the context of config module.

Reply:
To be clear you want the CLI opt to remain `:config` and then it just gets `Keyword.replace` called on it to be `:file`? Or you would like the opt to be `:file` everywhere? 

Do we just want to name the opt `:config_file` and call it that everywhere?

Reply:
Since opts in the context of check module may include lots of keys irrelevant to config loading, I was thinking to only pass the relevant options and to name them with relevance to the config loader. Therefore, it'd be something like:

```
{tools, config_opts} = Config.load(file: opts[:config])
```

This way you aren't forced to invent a name that makes sense in all layers, like your `:config_file` that makes the CLI option unnecessarily long since `--config` is pretty self explanatory in that layer, but may be misleading in the context of config loader where `:config` could mean anything

What do you think? Would you rather keep your `config_file` or go for shorter `config` renamed to `file` when passed to the loader? :)

Reply:
I can rename.

Reply:
Okay take a look. The name shuffle from `:config` to `:file` got a little wordy in `Check.run1` because of the `Config.load/1` taking more opts, so I took opportunity to drop `:config` there as well. This should be closer to what you're thinking, I think.